File: app/src/main/java/com/neome/api/meta/base/dto/StudioDtoDynamicRule.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Symbol;
import com.neome.api.meta.base.Types.EnumStudioCompType;
import com.neome.api.meta.base.Types.MetaIdFieldDynamicRule;
import com.neome.api.meta.base.Types.MetaIdVar;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class StudioDtoDynamicRule extends StudioBase
{
  @Nullable
  public StudioMapOfFieldDynamicCondition conditionNode;

  public EnumStudioCompType fieldType;

  public MetaIdFieldDynamicRule metaId;

  public Symbol name;

  @Nullable
  public MetaIdVar sourceVarId;
}

File: app/src/main/java/com/neome/api/meta/base/dto/StudioDtoDynamicRule.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumStudioCompType
import com.neome.api.meta.base.Types.MetaIdFieldDynamicRule
import com.neome.api.meta.base.Types.MetaIdVar
import com.neome.api.meta.base.dto.StudioBase
import com.neome.api.meta.base.dto.StudioMapOfFieldDynamicCondition
import com.neome.api.meta.base.Symbol

interface StudioDtoDynamicRule : StudioBase
{
  val conditionNode: StudioMapOfFieldDynamicCondition?
  val fieldType: EnumStudioCompType
  val metaId: MetaIdFieldDynamicRule
  val name: Symbol
  val sourceVarId: MetaIdVar?
}